text/0000-core-builder.md

```diff
@@ -0,0 +1,46 @@
+
+Users will be able to use the new builder to build and run their applications using any of the language family buildpacks included. Instead of requireing a decision for an optimised builder, they can "just" start and later decide on an optimised `run` image.
+
+## Motivation
+
+The Paketo project will be easier to use for users - they can start without being asked to decide on optimisation.
+The stacks and builders maintainer team will have less work to do - they will only need to maintain a single stack and builder instead of three.
+
+## Detailed Explanation
+
+We would replace the current `base`, `tiny` and `static` stacks with a single stack that includes all packages required by the buildpacks. Similarly, we would replace the current `base`, `tiny` and `static` builders with a single builder.
+
+## Rationale and Alternatives
+
+We could leave everything as it is. We would stick to maintaining four stacks repositories and eight builder repositories and creating new ones for every LTS we adopt. But that would increase the burden on the maintainers.
+
+## Implementation
+
+I would propose to create the new stack and builder as part of moving to Ubuntu Noble Numbat. I.e. 
+- we should create a repository `paketo-buildpacks/stack-ubuntu-noble`
+- we should take the `build` image content from the `paketo-buildpacks/noble-base-stack` repository and apply it to `paketo-buildpacks/stack-ubuntu-noble`
+- we should take the `run` image content from the `paketo-buildpacks/noble-base-stack` repository and apply it to `paketo-buildpacks/stack-ubuntu-noble`
+- we should take the `run` image content from the `paketo-buildpacks/noble-tiny-stack` repository and apply it to `paketo-buildpacks/stack-ubuntu-noble` as additional run image
+- we should take the `run` image content from the `paketo-buildpacks/noble-static-stack` repository and apply it to `paketo-buildpacks/stack-ubuntu-noble` as additional run image
+- we should create a repository `paketo-buildpacks/builder-ubuntu-noble` and another`paketo-buildpacks/builder-ubuntu-noble-buildpackless`
+    - we should use the above `build` image as a base layer for the `builder` image
+    - we should use the above `run` image as the default `run` image for the `builder.toml`
+    - we should use the above `run-static` and `run-tiny` images as additional `run` images for the `builder.toml`
+
+I would propose to delete the `paketo-buildpacks/noble-base-stack`, `paketo-buildpacks/noble-tiny-stack` and `paketo-buildpacks/noble-static-stack` repositories.
+I would propose to delete the `paketo-buildpacks/builder-noble-buildpackless-static`, `paketo-buildpacks/builder-noble-buildpackless-tiny` and `paketo-buildpacks/builder-noble-buildpackless-base` repositories.
+
+## Prior Art
+
+None.
+
+## Unresolved Questions and Bikeshedding
+
```

Comment:
Q: With pack, it is very easy to specify a different run image. Is it similarly easy when not using pack?

We have other platforms like Spring Boot users, Tekton users, and other CI systems where users don't use pack cli. Have we considered these to make sure that the user experience will still work well with this new approach?

Reply:
- Spring: https://docs.spring.io/spring-boot/maven-plugin/build-image.html#build-image.customization
- kpack: Doesn't support this setting explicitly (and unfortunately project.toml doesn't either)
- Tekton: https://github.com/buildpacks/tekton-integration/tree/main/task/buildpacks/0.4#parameters

Seems like we might want to raise this with CNB and work towards run image support in project.toml (which is supported by kpack and hence would solve this pretty generically).

Reply:
An alternative is the extension, which can choose which run image to use.

Reply:
> An alternative is the extension, which can choose which run image to use.

But doesn't have this the problem that user input (`--run-image`) would be overruled?

Reply:
Yes, good point. That's interesting, I dont know what happens in that case. I'll checkout it out, it might be something we need to clarify/patch on pack

Reply:
```suggestion
Q: With pack, it is very easy to specify a different run image. Is it similarly easy when not using pack?
A: We know this to work with some platforms. The notable exception is kpack.

For example:

- Spring: https://docs.spring.io/spring-boot/maven-plugin/build-image.html#build-image.customization
- Tekton: https://github.com/buildpacks/tekton-integration/tree/main/task/buildpacks/0.4#parameters
```